mitreattack/navlayers/generators/usage_generator.py

```diff
@@ -119,6 +123,10 @@ def generate_layer(self, match):
         """
         typeChecker(type(self).__name__, match, str, "match")
         raw_data, matched_obj = self.get_matrix_data(match)
+        if matched_obj.type == "attack-pattern":
+            print(f"Warning: The input match {match} corresponds with an ATT&CK Technique, which is not supported. "
+                  f"Please provide a group, software, or mitigation instead.")
+            raise StixObjectIsTechnique
```

Comment:
Instead of deny-listing `attack-pattern`s as a match type, it should instead allow-list the allowed types (group, software, mitigation). Currently you can still search for a tactic for instance and it'll crash trying to parse the tactic:

```
>>> y.generate_layer(match="Execution")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/jtrahan/working/workspaces/attack/mitreattack-python/mitreattack/navlayers/generators/usage_generator.py", line 125, in generate_layer
    raw_data, matched_obj = self.get_matrix_data(match)
  File "/Users/jtrahan/working/workspaces/attack/mitreattack-python/mitreattack/navlayers/generators/usage_generator.py", line 75, in get_matrix_data
    verb = 'mitigates' if obj.type == 'course-of-action' else 'uses'
AttributeError: 'dict' object has no attribute 'type'
```

This will improve resiliency as more STIX types are added to the knowledge base (Data Sources, Data Components, etc)

Reply:
Yeah, that's definitely a better approach; I'll make the substitution.